File: 01-ARRAYS/shuffle-the-array.php
<?php

# https://leetcode.com/problems/shuffle-the-array/
function shuffleArray(array &$nums, int $n): array
{
    $array = [];
    for ($i = 0; $i < $n; $i++) {
        $array[] = $nums[$i];
        $array[] = $nums[$i + $n];
    }
    return $array;
}

File: 01-ARRAYS/Go/shuffle-the-array.go
package main

func shuffle(nums []int, n int) []int {
	shuffled := make([]int, n*2)
	for i := 0; i < n*2; i += 2 {
		shuffled[i] = nums[i/2]
		shuffled[i+1] = nums[(i/2)+n]
	}

	return shuffled
}
